File: Request examples/request.py
import requests
from requests.structures import CaseInsensitiveDict

url = "http://192.168.2.142:23432/updater/post"

headers = CaseInsensitiveDict()
headers["API-key"] = "AAAAAAAAAAAAAAA"
headers["Content-Type"] = "application/json"

data = """
{
  "IP": "10.0.0.25",
  "DeviceID": "Device1"
}
"""

resp = requests.post(url, headers=headers, data=data)

print(resp.status_code)

File: Request examples/request.php
<?php

$url = "http://192.168.2.142:23432/updater/post";

$curl = curl_init($url);
curl_setopt($curl, CURLOPT_URL, $url);
curl_setopt($curl, CURLOPT_POST, true);
curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);

$headers = array(
   "API-key: AAAAAAAAAAAAAAA",
   "Content-Type: application/json",
);
curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);

$data = <<<DATA
{
  "IP": "10.0.0.25",
  "DeviceID": "Device1"
}
DATA;

curl_setopt($curl, CURLOPT_POSTFIELDS, $data);

//for debug only!
curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);

$resp = curl_exec($curl);
curl_close($curl);
var_dump($resp);

?>